test/lsp_test_runner.cc

```diff
@@ -139,11 +139,15 @@ void validateCodeActions(LSPWrapper &lspWrapper, Expectations &test, string file
             continue;
         }
 
-        bool codeActionTitleUnique =
-            receivedCodeActionsByTitle.find(codeAction->title) == receivedCodeActionsByTitle.end();
-        CHECK_MESSAGE(codeActionTitleUnique, "Found code action with duplicate title: " << codeAction->title);
-
-        if (codeActionTitleUnique) {
+        // "Move method" refactorings always have the same title
+        if (codeAction->kind != CodeActionKind::RefactorExtract) {
+            bool codeActionTitleUnique =
+                receivedCodeActionsByTitle.find(codeAction->title) == receivedCodeActionsByTitle.end();
+            CHECK_MESSAGE(codeActionTitleUnique, "Found code action with duplicate title: " << codeAction->title);
```

Comment:
Isn't this guard required for the correctness of the test runner itself?

If `codeAction->kind` is `RefactorExtract` and there is already one of them, then we're going to overwrite the previous one on line 151. Is that not a problem? Not really sure how `receivedCodeActionsByTitle` is being used.

Reply:
> Isn't this guard required for the correctness of the test runner itself?

Yep, that's true. The guard rail was brought up for the quickfix assertions. For quickfixes you don't want to have two fixes at the same place with the same title. 

But that assumption is not correct for refactor actions. For example, you can have two refactor code actions with the same title `Move method` in one location if that location is `prop :foo, Integer`. The prop test was the main driver for introducing this change

You also right about code actions overwriting each other, but that's fine for now, because the only case where the action is overwritten is the same `test/testdata/lsp/code_actions/move_method/prop.rb` test. And the refactor actions in the test are no-ops, so that's okay.

My goal here was to get the prop test working without large refactors to the test runner.

Reply:
In that case can you leave a TODO comment in the test runner? this seems like it could be a footgun in the future.

Reply:
Fortunately, after disabling the refactoring for the instance methods we don't need this hack anymore